src/mon/OSDMonitor.cc

```diff
@@ -957,6 +957,67 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 			  << "required " << ceph_release_name(mv);
 	pending_inc.new_require_min_compat_client = mv;
       }
+
+      // auto-enable pool applications upon upgrade
+      // NOTE: this can be removed post-Luminous assuming upgrades need to
+      // proceed through Luminous
+      if (osdmap.require_osd_release < CEPH_RELEASE_LUMINOUS) {
+        for (auto &pool_pair : tmp.pools) {
+          int64_t pool_id = pool_pair.first;
+          pg_pool_t pg_pool = pool_pair.second;
+          std::string pool_name = tmp.get_pool_name(pool_id);
+          bool updated = false;
+
+          // CephFS
+          FSMap const &pending_fsmap = mon->mdsmon()->get_pending();
+          if (pending_fsmap.pool_in_use(pool_id)) {
+            dout(10) << __func__ << " auto-enabling CephFS on pool '"
+                     << pool_name << "'" << dendl;
+            pg_pool.application_metadata.insert(
+              {pg_pool_t::APPLICATION_NAME_CEPHFS, {}});
+            updated = true;
+          }
+
+          // RBD heuristics (default OpenStack pool names)
+          if (boost::algorithm::contains(pool_name, "rbd") ||
+              boost::algorithm::contains(pool_name, "images") ||
+              boost::algorithm::contains(pool_name, "volumes") ||
+              boost::algorithm::contains(pool_name, "backups") ||
+              boost::algorithm::contains(pool_name, "vms")) {
+            dout(10) << __func__ << " auto-enabling RBD on pool '"
+                     << pool_name << "'" << dendl;
+            pg_pool.application_metadata.insert(
+              {pg_pool_t::APPLICATION_NAME_RBD, {}});
+            updated = true;
+          }
+
+          // RGW heuristics
+          if (boost::algorithm::contains(pool_name, ".rgw") ||
+              boost::algorithm::contains(pool_name, ".log") ||
+              boost::algorithm::contains(pool_name, ".intent-log") ||
+              boost::algorithm::contains(pool_name, ".usage") ||
+              boost::algorithm::contains(pool_name, ".users")) {
+            dout(10) << __func__ << " auto-enabling RGW on pool '"
+                     << pool_name << "'" << dendl;
+            pg_pool.application_metadata.insert(
+              {pg_pool_t::APPLICATION_NAME_RGW, {}});
+            updated = true;
+          }
+
+          // OpenStack gnocchi
```

Comment:
Why are we singling out gnocchi as a special thing we recognize? I don't think it belongs here, especially with a fairly common string like "metrics".